teamcity: failed tests on master: acceptance/TestDockerCLI

The following tests appear to have failed:

[#559666](https://teamcity.cockroachdb.com/viewLog.html?buildId=559666):


```
--- FAIL: acceptance/TestDockerCLI (419.560s)
test_log_scope.go:81: test logs captured to: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/logTestDockerCLI859470117
test_log_scope.go:62: use -show-logs to present logs inline
--- FAIL: acceptance/TestDockerCLI: TestDockerCLI/test_audit_log.tcl (36.280s)

--- FAIL: acceptance/TestDockerCLI (419.560s)
test_log_scope.go:81: test logs captured to: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/logTestDockerCLI859470117
test_log_scope.go:62: use -show-logs to present logs inline
--- FAIL: acceptance/TestDockerCLI: TestDockerCLI/test_audit_log.tcl (36.280s)


```

Please assign, take a look and update the issue accordingly.

--- a/pkg/cli/interactive_tests/common.tcl
+++ b/pkg/cli/interactive_tests/common.tcl
@@ -107,6 +107,18 @@ proc stop_server {argv} {
     report "END STOP SERVER"
 }
 
+proc rotate_server_logs {} {
+    report "BEGIN ROTATE LOGS"
+    system "kill -USR2 `cat server_pid` 2>/dev/null"
+    report "END ROTATE LOGS"
+}
+
+proc flush_server_logs {} {
+    report "BEGIN FLUSH LOGS"
+    system "kill -USR1 `cat server_pid` 2>/dev/null"
+    report "END FLUSH LOGS"
+}
+
 proc force_stop_server {argv} {
     report "BEGIN FORCE STOP SERVER"
     system "$argv quit & sleep 1; if kill -CONT `cat server_pid` 2>/dev/null; then kill -TERM `cat server_pid`; sleep 1; if kill -CONT `cat server_pid` 2>/dev/null; then kill -KILL `cat server_pid`; fi; fi"

--- a/pkg/cli/interactive_tests/test_audit_log.tcl
+++ b/pkg/cli/interactive_tests/test_audit_log.tcl
@@ -43,14 +43,10 @@ end_test
 
 # force rotate the logs: the test below must not see the
 # log entries that were already generated above.
-force_stop_server $argv
-start_server $argv
+rotate_server_logs
 
-send "\r"
-eexpect "opening new connection"
-eexpect root@
-send "\r"
-eexpect "t>"
+# Check the log indeed is empty
+system "if grep -q helloworld $logfile; then false; fi"
 
 start_test "Check that audit removal is logged too"
 send "ALTER TABLE helloworld EXPERIMENTAL_AUDIT SET OFF;\r"

--- a/pkg/cli/interactive_tests/test_exec_log.tcl
+++ b/pkg/cli/interactive_tests/test_exec_log.tcl
@@ -1,0 +1,48 @@
+#! /usr/bin/env expect -f
+
+source [file join [file dirname $argv0] common.tcl]
+
+start_server $argv
+
+spawn $argv sql
+eexpect root@
+
+set logfile logs/db/logs/cockroach-sql-exec.log
+
+start_test "Check that the exec log is not created by default"
+system "if test -e $logfile; then false; fi"
+end_test
+
+start_test "Check that the exec log is created after enabled"
+send "SET CLUSTER SETTING sql.trace.log_statement_execute = TRUE;\r"
+eexpect root@
+system "test -e $logfile"
+end_test
+
+start_test "Check that statements get logged to the exec log but stop logging when disabled"
+send "SELECT 'helloworld';\r"
+eexpect root@
+
+# Errors must be logged too
+send "SELECT nonexistent;\r"
+eexpect "column name"
+eexpect "nonexistent"
+eexpect "not found"
+eexpect root@
+
+# Check logging after disable
+send "SET CLUSTER SETTING sql.trace.log_statement_execute = FALSE;\r"
+eexpect root@
+send "SELECT 'lovely';\r"
+eexpect root@
+
+flush_server_logs
+
+# Now check the items are there in the log file.
+system "grep -q helloworld $logfile"
+system "grep -q nonexistent $logfile"
+system "if grep -q lovely $logfile; then false; fi"
+
+end_test
+
+stop_server $argv

--- a/pkg/cli/interactive_tests/test_rotate_flush.tcl
+++ b/pkg/cli/interactive_tests/test_rotate_flush.tcl
@@ -1,0 +1,24 @@
+#! /usr/bin/env expect -f
+
+source [file join [file dirname $argv0] common.tcl]
+
+set logfile logs/db/logs/cockroach.log
+
+start_server $argv
+
+start_test "Check that SIGUSR1 says it flushes the log"
+flush_server_logs
+system "grep 'signal.*received, flushing logs' $logfile"
+end_test
+
+start_test "Check that SIGUSR2 flushes and rotates the logs"
+rotate_server_logs
+# The log have been rotated so the flush/rotate messages should not be there any more.
+system "if grep 'signal.*received, flushing logs' $logfile; then false; fi"
+system "if grep 'signal.*received, rotating logs' $logfile; then false; fi"
+# However they must be in the directory in some previous log file.
+system "grep 'signal.*received, flushing logs' logs/db/logs/*.log"
+system "grep 'signal.*received, rotating logs' logs/db/logs/*.log"
+end_test
+
+stop_server $argv

--- a/pkg/util/log/clog.go
+++ b/pkg/util/log/clog.go
@@ -27,6 +27,7 @@ import (
 	stdLog "log"
 	"math"
 	"os"
+	"os/signal"
 	"path/filepath"
 	"regexp"
 	"runtime"
@@ -34,6 +35,7 @@ import (
 	"strconv"
 	"strings"
 	"sync/atomic"
+	"syscall"
 	"time"
 
 	"github.com/cockroachdb/cockroach/pkg/build"
@@ -588,6 +590,22 @@ func init() {
 	logging.fatalCh = make(chan struct{})
 
 	go flushDaemon()
+	go signalFlusher()
+}
+
+// signalFlusher flushes the log(s) every time SIGUSR1 is received.
+func signalFlusher() {
+	flushCh := make(chan os.Signal, 1)
+	signal.Notify(flushCh, syscall.SIGUSR1, syscall.SIGUSR2)
+
+	for sig := range flushCh {
+		Infof(context.Background(), "%s received, flushing logs", sig)
+		Flush()
+		if sig == syscall.SIGUSR2 {
+			Infof(context.Background(), "%s received, rotating logs", sig)
+			Rotate()
+		}
+	}
 }
 
 // LoggingToStderr returns true if log messages of the given severity
@@ -611,6 +629,16 @@ func Flush() {
 	for _, l := range secondaryLogRegistry.mu.loggers {
 		// Some loggers (e.g. the audit log) want to keep all the files.
 		l.logger.lockAndFlushAll()
+	}
+}
+
+// Rotate rotates all the currently open log files.
+func Rotate() {
+	logging.lockAndRotateFile()
+	secondaryLogRegistry.mu.Lock()
+	defer secondaryLogRegistry.mu.Unlock()
+	for _, l := range secondaryLogRegistry.mu.loggers {
+		l.logger.lockAndRotateFile()
 	}
 }
 
@@ -964,6 +992,16 @@ func (l *loggingT) exitLocked(err error) {
 	}
 	l.flushAndSync(true /*doSync*/)
 	l.exitFunc(2)
+}
+
+func (l *loggingT) lockAndRotateFile() {
+	l.mu.Lock()
+	defer l.mu.Unlock()
+	if sb, ok := l.file.(*syncBuffer); ok {
+		if err := sb.rotateFile(timeutil.Now()); err != nil {
+			l.exitLocked(err)
+		}
+	}
 }
 
 // syncBuffer joins a bufio.Writer to its underlying file, providing access to the

